src/testRunner/unittests/tsbuild/moduleResolution.ts

```diff
@@ -86,4 +86,36 @@ namespace ts.tscWatch {
             commandLineArgs: ["-b", "/src/packages/pkg1.tsconfig.json", "/src/packages/pkg2.tsconfig.json", "--verbose", "--traceResolution"],
         });
     });
-}
+
+    describe("unittests:: tsbuild:: moduleResolution:: impliedNodeFormat differs between projects for shared file", () => {

```

Comment:
The options that are marked as `affectsModuleResolution` wouldnt have issues in general because tsbuild does not cache sourceFiles as of now beyond current build time.. So program is never reused and the resolution is done from scratch.
The impliedNodeFormat was the only thing that mattered because that field gets stuffed onto source file and is not updated at all. Infact i changed my fix to set it on the source File just like we set the other properties to ensure they are correct (like path, resolvedPath, packageJson locations etc)

Btw we have a test when two projects differ in module resolution option
Eg https://github.com/microsoft/TypeScript/blob/main/src/testRunner/unittests/tsbuild/moduleResolution.ts#L69
https://github.com/microsoft/TypeScript/blob/main/src/testRunner/unittests/tsbuild/transitiveReferences.ts#L31

Reply:
> The options that are marked as `affectsModuleResolution` wouldnt have issues in general because tsbuild does not cache sourceFiles as of now beyond current build time

But don't the `module` and `moduleResolution` options have `affectsModuleResolution`? Shouldn't they have not had issues? Maybe I'm missing something.

Reply:
Okay, I just read the fix - let me think about it a bit more.